tests/kernel/sched/metairq/src/main.c

```diff
@@ -59,7 +59,7 @@ void metairq_thread(void)
 	printk("give sem1\n");
 	k_sem_give(&coop_sem1);
 
-	printk("metairq end\n");
+	printk("metairq end, should switch back to co-op thread2\n");
```

Comment:
Seems like the changes to this test are just extra verbosity and a new comment to explain the sequencing?  Could remove, or put in a separate commit.

Reply:
agreed will split this out. I added this stuff because one version of this PR failed this test, and I had to spend some time trying to figure out how this test was supposed to behave.